### ISSUE TITLE
Add link identifier support for the /taxon/ service

### ISSUE THREAD

**Issue by djtfmartin** [AtlasOfLivingAustralia/bie-index#11]
This is to enable lookups like:

/taxon/Macropus+rufus

This is done in the old BIE by doing a pass of binonials and common names and creating a "linkIdentifier" where there is a guarantee of uniqueness for the name -> taxon within the dataset.

**Comment by adam-collins** [comment]
/species/Macropus+rufus works
/guid/Macropus%20rufus works
/guid/Macropus+rufus fails

**Comment by adam-collins** [comment]
1. `/species/{name}` calls `regularise(name)` replaces `+` with ' ' using `URLDecoder.decode(name)`. It does more varied searching than `/guid/{name}`
2. `/guid/{name}`, possibly all services, makes use of `Encoder.escapeSolr(name)`. This escapes the `+`. It only looks at 3 fields `scientificName`, `nameComplete` and `commonName`.

I do not want to add `regularise` to `/guid/{name}` because then you cannot search for names containing a `+` e.g. https://bie-ws.ala.org.au/ws/guid/Dimeria%20sp.%20Mosquito%20Point%20(J.R.Clarkson+%209994) without removing the `escapeSolr` and requiring requests to escape for SOLR. Knowledge of SOLR escaping should not be required to use this service.

In conclusion, making no change.
